File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    
    let max = 0;
    const set = new Set(nums);
    
    for (let num of set){
        if (set.has(num - 1)) continue;
        
        let curr = num
        let currMax = 1
        
        while (set.has(curr + 1)){
            curr++
            currMax++
        }
        max = Math.max(max, currMax)
    }
    
    return max
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        nums = sorted(set(nums))
        count = 0
        max_count = 0
        previous = 0
        
        for num in nums:
            if count == 0:
                count += 1
                prev = num
            elif num == prev + 1:
                count += 1
                prev = num
            else:
                if count > max_count:
                    max_count = 0
                    max_count += count
                count = 1
                prev = num
        if count > max_count:
            return count
        return max_count